### PR TITLE
[SIM_FLUTTER-88] [Integrate] User can logout

### PR DESCRIPTION
**Pull request by CaneteJoash**
#### **Backlog Link:**
- https://framgiaph.backlog.com/view/SIM_FLUTTER-91
#### Slack link
- https://sun-philippine.slack.com/archives/C059L7XAXFB/p1687498995617439
#### Definition of Done
- [x] should display login page after logging out 

#### Note
- Name is still static. it can be dynamic [here ](https://framgiaph.backlog.com/view/SIM_FLUTTER-61)

### CODE REVIEW

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #26 

Working: 
- Able to logout successfully
- Once name is clicked, logout button will appear

https://github.com/framgia/sph-flutter/assets/86758474/2331c636-274a-4257-b99f-2ab634dffc8a

Note:
- Dynamic name will be apply on this [task](https://framgiaph.backlog.com/view/SIM_FLUTTER-61)
- If the user logs in, then goes back to the CP homescreen, and then goes back to the application. User should stay logged in. - Will create a task for this

Testing Hours: 0.50
Comment: 0
Iteration: 0

File: frontend/lib/src/components/auth/auth_header.dart [hunk 1]
```diff
@@ -20,6 +29,39 @@ class AuthHeader extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    const storage = FlutterSecureStorage();
+    final networkConfig = NetworkConfig();
+
+    void dropdownCallback(selectedValue) async {
+      if (selectedValue == 'logout') {
+        // note: csrf is for web, web implementation not thoroughly tested
+        await networkConfig.getCsrftoken();
```
Comment (jstephend-sun) [inline frontend/lib/src/components/auth/auth_header.dart]: any particular reason why the csrf token is retrieved again here?

File: frontend/lib/src/components/auth/auth_header.dart [hunk 2]
```diff
@@ -20,6 +29,39 @@ class AuthHeader extends StatelessWidget {
...
+        final logoutResponse = await client.post(
+          logoutUrl,
+        );
+
+        if (logoutResponse.statusCode == HttpStatus.ok) {
+          final message = logoutResponse.data['data']['message'];
+
+          // delete the stored login key
+          await storage.delete(key: loginTokenKey);
+          Get.to(() => const LoginPage());
```
Comment (jstephend-sun) [inline frontend/lib/src/components/auth/auth_header.dart]: get.to is incorrect here, triggering the back functionality on the phone after logging out goes back to user list page

File: frontend/lib/src/helper/snackbar/show_snackbar.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+
+/*
+  Reusable snackbar dialog
```
Comment (jstephend-sun) [inline frontend/lib/src/helper/snackbar/show_snackbar.dart]:
```suggestion
  Reusable snackbar dialog?
```

does it have dialog functionality? user can click cancel or confirm button?

File: frontend/lib/src/helper/snackbar/show_snackbar.dart [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  @param title: String value to display the title of the alert dialog
```
Comment (jstephend-sun) [inline frontend/lib/src/helper/snackbar/show_snackbar.dart]:
```suggestion
  @param title: String value to display the title of the alert dialog??
```

it this alert or snackbar?

File: frontend/lib/src/helper/snackbar/show_snackbar.dart [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  @param content: String value to display the content of the alert dialog
```
Comment (jstephend-sun) [inline frontend/lib/src/helper/snackbar/show_snackbar.dart]:
```suggestion
  @param content: String value to display the content of the alert dialog
```
same here

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changes #26

review time: 10mins
comments: 5

**Comment by Dbulawan13** [comment]
Comment:

- [ ] Should retain in the login page
Steps:
- login
- logout
- press back button/swipe to trigger back function

https://github.com/framgia/sph-flutter/assets/86758474/7e17c69c-210d-4819-9ace-de48da3c215f

Comment: 1

**Review by Dbulawan13 — APPROVED** [review]
Approve PR #26 

Working: 
- Able to logout successfully
- Once name is clicked, logout button will appear

https://github.com/framgia/sph-flutter/assets/86758474/45920741-8632-4b60-baea-13d44e589977

Note:
- Dynamic name will be apply on this [task](https://framgiaph.backlog.com/view/SIM_FLUTTER-61)
- If the user logs in, then goes back to the CP homescreen, and then goes back to the application. User should stay logged in. - Will create a task for this

Testing Hours: 1
Comment: 1
Iteration: 1

**Review by jstephend-sun — APPROVED** [review]
Approve #26

review time: 10mins
comments: 5

**Pull request merged**